pkg/tanka/find.go

```diff
@@ -2,123 +2,165 @@ package tanka
 
 import (
 	"fmt"
-	"os"
 	"path/filepath"
+	"runtime"
+	"time"
 
+	"github.com/grafana/tanka/pkg/jsonnet"
 	"github.com/grafana/tanka/pkg/jsonnet/jpath"
 	"github.com/grafana/tanka/pkg/spec/v1alpha1"
 	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
 	"k8s.io/apimachinery/pkg/labels"
 )
 
 // FindOpts are optional arguments for FindEnvs
 type FindOpts struct {
 	JsonnetOpts
-	Selector labels.Selector
+	Selector    labels.Selector
+	Parallelism int
 }
 
 // FindEnvs returns metadata of all environments recursively found in 'path'.
 // Each directory is tested and included if it is a valid environment, either
 // static or inline. If a directory is a valid environment, its subdirectories
 // are not checked.
 func FindEnvs(path string, opts FindOpts) ([]*v1alpha1.Environment, error) {
-	// find all environments at dir
-	envs, errs := find(path, Opts{JsonnetOpts: opts.JsonnetOpts})
-	if errs != nil {
-		return envs, ErrParallel{errors: errs}
-	}
-
-	// optionally filter
-	if opts.Selector == nil || opts.Selector.Empty() {
-		return envs, nil
-	}
+	return findEnvsFromPaths([]string{path}, opts)
+}
 
-	filtered := make([]*v1alpha1.Environment, 0, len(envs))
-	for _, e := range envs {
-		if !opts.Selector.Matches(e.Metadata) {
-			continue
-		}
-		filtered = append(filtered, e)
-	}
+// FindEnvsFromPaths does the same as FindEnvs but takes a list of paths instead
+func FindEnvsFromPaths(paths []string, opts FindOpts) ([]*v1alpha1.Environment, error) {
+	return findEnvsFromPaths(paths, opts)
+}
 
-	return filtered, nil
+type findJsonnetFilesOut struct {
+	jsonnetFiles []string
+	err          error
 }
 
-func findErr(path string, err error) []error {
-	return []error{fmt.Errorf("%s:\n %w", path, err)}
+type findEnvsOut struct {
+	envs []*v1alpha1.Environment
+	err  error
 }
 
-// find implements the actual functionality described at 'FindEnvs'
-func find(path string, opts Opts) ([]*v1alpha1.Environment, []error) {
-	// try if this has envs
-	list, err := List(path, opts)
-	if err != nil &&
-		// expected when looking for environments
-		!errors.As(err, &jpath.ErrorNoBase{}) &&
-		!errors.As(err, &jpath.ErrorFileNotFound{}) {
-		return nil, findErr(path, err)
-	}
-	if len(list) != 0 {
-		// it has. don't search deeper
-		return list, nil
+func findEnvsFromPaths(paths []string, opts FindOpts) ([]*v1alpha1.Environment, error) {
+	if opts.Parallelism >= 0 {
+		opts.Parallelism = runtime.NumCPU()
```

Comment:
This is essentially overriding the value - is that what you're intending to do? Do you mean `== 0`?

Reply:
🤦 Coding right before bed, must've had blurry vision 😬